examples_test.go

```diff
@@ -67,3 +67,25 @@ func ExampleScan_parseDuration() {
 	// Output:
 	// 3m20s
 }
+
+func ExampleScan_repeatedly() {
+	line := []byte("www.google.com:1234 www.google.com:2345")
+	r := regexp.MustCompile(`(\S+):(\d+)`)
+
+	for {
+		var (
+			position re.Position
+			host     string
+			port     int
+		)
+		err := re.Scan(r, line, &position, &host, &port)
```

Comment:
My intention with this suggestion was that re.Position would not be special. It can occur anywhere in the list of args and matches a capture group. So for example, a user could place an empty capture group at the end:

```
   `(\S+):(\d+)()` // args will be &host, &port, &pos
```

Or capture the whole thing:

```
   `((\S+):(\d+))  // args will be &pos, &host, &port
```

Reply:
My intention was that you could capture Range of the whole regexp without having to actually change how you wrote the regexp.

It might also be useful to capture Range of individual matching groups, but we can do that by supporting additional types of parsing functions. If we add support for func([]byte, Range) error, then we can define our first custom parsing function:

```
func GroupPosition(value interface{}, range *Range) func([]byte, Range) error {
  return func(content []byte, r Range) error {
   *range = r
   return assign(value, content)
  }
}

```
and get the range for a matching group as follows:

```
exp := regexp.MustCompile(`(\S+):(\d+)`)
re.Scan(exp, data, re.GroupPosition(&host, &hostRange), &port)
```

(I haven't thought much about the name for GroupPosition yet)

Reply:
I've implemented `func(content []byte, r Range) error`, but haven't written the GroupPosition function yet. I'm not sure this is a common thing to need, so I'm not sure whether we should provide the GroupPosition function by default. Maybe we should use this support to implement a `Maybe`, though, which would have similar functionality to C++ RE2's support of std::optional.

Reply:
Please see my other comment. I would like add just one new type that captures a sub-range of the match like other parameters.